src/vm/interoputil.cpp

```diff
@@ -1618,6 +1619,54 @@ BOOL CanCastComObject(OBJECTREF obj, MethodTable * pTargetMT)
     }
 }
 
+// Returns TRUE iff the argument represents the "__ComObject" type or
+// any type derived from it (i.e. typelib-imported RCWs).
+BOOL IsComWrapperClass(TypeHandle type)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+    }
+    CONTRACTL_END;
+    
+    MethodTable* pMT = type.GetMethodTable();
+    if (pMT == NULL)
+        return FALSE;
+        
+    return pMT->IsComObjectType();
+}
+
+// Returns TRUE iff the argument represents the "__ComObject" type.
+BOOL IsComObjectClass(TypeHandle type)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+        SO_TOLERANT;
+    }
+    CONTRACTL_END;
+
+    if (!type.IsTypeDesc())
+    {
+        MethodTable *pMT = type.AsMethodTable();
+
+        if (pMT->IsComObjectType())
+        {
+            // May be __ComObject or typed RCW. __ComObject must have already been loaded
+            // if we see an MT marked like this so calling the *NoInit method is sufficient.
+#ifdef FEATURE_COMINTEROP
```

Comment:
The whole method is always going to return FALSE. You can make the ifdef wider.

Reply:
Fixed Update